### PR TITLE
CEPHFS - Tier3 - CEPH-83584084

### PR DESCRIPTION
**Pull request by hkadam134**
Polarion TC : [CEPH-83584084 - Validate if no repeated mclientscaps(revoke) entries in MDS which was causing the Ceph Status to go Unhealthy](https://polarion.engineering.redhat.com/polarion/redirect/project/CEPH/workitem?id=CEPH-83584084)

Logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-MTZI8T/

====================================================

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/cephfs/cephfs_bugs/test_validate_mclientcaps_revoke_ack.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+            dst=f"/root/{create_large_file_script}",
+        )
+        clients[0].exec_command(
+            sudo=True,
+            cmd=f"python /root/{create_large_file_script} {' '.join(file_paths)}",
+            timeout=7200,
+        )
+
+        log.info(
+            "Check for the Ceph Health to see if there are any deadlock bw unlink and rename."
```
Comment (neha-gangadhar) [inline tests/cephfs/cephfs_bugs/test_validate_mclientcaps_revoke_ack.py:159]: how are we ensuring unlink and rename ops are triggered?

File: tests/cephfs/cephfs_bugs/test_validate_mclientcaps_revoke_ack.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+                f"This test requires minimum 1 client nodes.This has only {len(clients)} clients"
+            )
+            return 1
+        fs_util_v1.prepare_clients(clients, build)
+        fs_util_v1.auth_list(clients)
+        default_fs = "cephfs"
+        subvolume_group_name = "subvol_group1"
+        subvolume_name = "subvol"
+
+        clients[0].exec_command(
```
Comment (neha-gangadhar) [inline tests/cephfs/cephfs_bugs/test_validate_mclientcaps_revoke_ack.py:51]: do we not have a generic method in utils to set/get config values?  If not good to add and use as a generic method as many tests will be needing the method. Looks good for the test, just food for thought! we should create reusable code.

File: tests/cephfs/cephfs_bugs/test_validate_mclientcaps_revoke_ack.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+            sudo=True,
+            cmd="ceph config set mds log_to_file true",
+        )
+        out, rc = clients[0].exec_command(
+            sudo=True,
+            cmd="ceph config get mds log_to_file",
+        )
```
Comment (AmarnatReddy) [inline tests/cephfs/cephfs_bugs/test_validate_mclientcaps_revoke_ack.py:58]: This way of setting was not logging the things to log file on mds nodes Hemanth. 
Was this working now ?

File: tests/cephfs/cephfs_bugs/test_validate_mclientcaps_revoke_ack.py [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+                    cmd=f"grep -i 'mclientcaps(revoke)'  /var/log/ceph/{fsid}/ceph-mds*",
+                    check_ec=False,
+                )
+                log.info(out)
+                if "mclientcaps(revoke)" not in out:
+                    print(
+                        "Test is passed: No repeated mclientcaps Revoke entries found in MDS log."
+                    )
+                else:
+                    log.error(
```
Comment (sumabai) [inline tests/cephfs/cephfs_bugs/test_validate_mclientcaps_revoke_ack.py:181]: I have seen that with log.error executed, Test case is not marked Failed as return 0 happens.
with "mclientcaps(revoke)" in out, in else block we should add return 1 or we can raise an exception?

File: tests/cephfs/cephfs_bugs/test_validate_mclientcaps_revoke_ack.py (hunk 1 above)
Comment (hkadam134) [reply tests/cephfs/cephfs_bugs/test_validate_mclientcaps_revoke_ack.py:159]: Sorry, the sentence is wrong. We are just checking the Health is OK, if anything other than OK , we are raising exception in "def get_ceph_health_status"

File: tests/cephfs/cephfs_bugs/test_validate_mclientcaps_revoke_ack.py (hunk 2 above)
Comment (hkadam134) [reply tests/cephfs/cephfs_bugs/test_validate_mclientcaps_revoke_ack.py:51]: Done

File: tests/cephfs/cephfs_bugs/test_validate_mclientcaps_revoke_ack.py (hunk 4 above)
Comment (hkadam134) [reply tests/cephfs/cephfs_bugs/test_validate_mclientcaps_revoke_ack.py:181]: Added CommandFailed exception

**Review by sumabai — APPROVED** [review]
lgtm

**Review by AmarnatReddy — APPROVED** [review]
LGTM

**Pull request merged**